public/app/plugins/datasource/tempo/QueryField.tsx

```diff
@@ -16,16 +16,27 @@ import { TraceToLogsOptions } from 'app/core/components/TraceToLogsSettings';
 import React from 'react';
 import { LokiQueryField } from '../loki/components/LokiQueryField';
 import { TempoDatasource, TempoQuery, TempoQueryType } from './datasource';
+import LokiDatasource from '../loki/datasource';
+import { LokiQuery } from '../loki/types';
+import { PrometheusDatasource } from '../prometheus/datasource';
+import useAsync from 'react-use/lib/useAsync';
 
 interface Props extends ExploreQueryFieldProps<TempoDatasource, TempoQuery>, Themeable2 {}
 
 const DEFAULT_QUERY_TYPE: TempoQueryType = 'traceId';
+
 interface State {
-  linkedDatasource?: DataSourceApi;
+  linkedDatasourceUid?: string;
+  linkedDatasource?: LokiDatasource;
```

Comment:
Not exactly, the picker only allows Loki to be selected https://github.com/grafana/grafana/blob/c02ead911345516bc2a4621e86da0a22efe7c37d/public/app/core/components/TraceToLogsSettings.tsx#L39.

In general this is not ideal, but if we wanted to also allow elastic we would also need to implement the querying like this https://github.com/grafana/grafana/blob/4d7af6d703608bd06029aec04d3cce07475057f6/public/app/plugins/datasource/tempo/datasource.ts#L57.

In the context of Glue, hard to say how much it could be simplified, ideally you could link any data source without Tempo data source having to know/understand the target data source but not sure how to do that.